index.bs

```diff
@@ -1739,6 +1781,171 @@ contexts that have active documents; navigation can also cause contexts to
 become inaccessible but not yet get discarded because bfcache.
 </div>
 
+## Log ## {#module-log}
+
+The <dfn export for=modules>log</dfn> module contains functionality and events
+related to logging.
+
+### Definition ### {#module-log-definition}
+
+[=remote end definition=]
+
+<pre class="cddl remote-cddl">
+
+Command = (GetCommand)
+</pre>
+
+[=local end definition=]
+
+<pre class="cddl local-cddl">
+
+LogEvent = (
+  LogEntryAddedEvent
+)
+</pre>
+
+### Types ### {#module-log-types}
+
+#### Log Entry #### {#types-log-log-entry}
+
+<pre class="cddl local-cddl">
+
+LogLevel = "debug" / "info" / "warning" / "error"
```

Comment:
All these entries are related to console API calls, but how would we handle Javascript errors and warnings as raised by the browser? Do we also want to have that as a separate event like `Runtime.exceptionThrown` vs `Runtime.consoleAPICalled` in CDP?

Also there are way more types available like `dir`, `timeEnd`, and many others. Do we want to fold all of these into the `info` level? It will make it impossible to watch for these specific events.

Reply:
The additional detail is provided on the extra field; the `ConsoleLogExtra` has a `type` field that allows all the detail for the different console types. I think having a very flat design in which any log type could add to the top-level log level would make it very hard to extend because clients would get top-level types they didn't understand and that don't really fit into the "give me all events above level X" paradigm.

Although we haven't designed filtering yet, I agree that it's going to be harder to filter on extra data. In practice for console events in tests the author is probably pretty in control of what's emitted anyway, so hopefully just being able to filter by type and level is enough.

Reply:
So what about Javascript exceptions then? Where would those fold into? Are those equal to `console.error()` calls?

Reply:
No, this is the difference between `type: console` and `type: javascript` at the top level.